.github/workflows/release.yml

```diff
@@ -0,0 +1,114 @@
+        description: 'Branch to release (e.g., main, release-1.0)'
+        default: 'main'
+      run_tests:
+        description: 'Run tests before packaging and releasing (true/false)'
+        default: 'true'
+
+jobs:
+  build:
+    name: Build and Release (${{ matrix.arch }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            arch: linux
+          - os: macos-latest
+            arch: macos
+          - os: windows-latest
+            arch: windows
+      max-parallel: 3
+
+    steps:
+      - name: Check if architecture is selected
+        id: check_arch
+        shell: bash
+        run: |
+          echo "Selected architectures: ${{ github.event.inputs.architectures }}"
+          if [[ "${{ github.event.inputs.architectures }}" == *"${{ matrix.arch }}"* ]]; then
+            echo "should_build=true" >> $GITHUB_OUTPUT
+          else
+            echo "should_build=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Skip job if architecture not selected
+        if: ${{ steps.check_arch.outputs.should_build == 'false' }}
+        run: |
+          echo "Skipping build for architecture ${{ matrix.arch }}"
+          exit 0
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.release_branch || github.ref }}
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: "vscode/.nvmrc"
+
+      - name: Install dependencies
+        working-directory: ./vscode
+        run: npm ci
+
+      - name: Lint sources
+        working-directory: ./vscode
+        run: npm run lint
+
+      - name: Build
+        working-directory: ./vscode
+        run: npm run package
+
+      # Install Xvfb on Linux
+      - name: Install Xvfb (Linux)
+        if: ${{ runner.os == 'Linux' }}
+        run: sudo apt-get install -y xvfb
+
+      # Run tests on Linux
+      - name: Run tests (Linux)
+        if: ${{ github.event.inputs.run_tests == 'true' && runner.os == 'Linux' }}
+        working-directory: ./vscode
+        run: xvfb-run npm test
+        shell: bash
+
+      # Run tests on macOS
+      - name: Run tests (macOS)
+        if: ${{ github.event.inputs.run_tests == 'true' && runner.os == 'macOS' }}
+        working-directory: ./vscode
+        run: npm test
+        shell: bash
+
+      # Run tests on Windows
+      - name: Run tests (Windows)
+        if: ${{ github.event.inputs.run_tests == 'true' && runner.os == 'Windows' }}
+        working-directory: ./vscode
+        run: npm test
+        shell: cmd
+
+      - name: Generate .vsix package
+        working-directory: ./vscode
+        run: |
+          npm install @vscode/vsce
+          npx vsce package
+
+      - name: Upload VSIX artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: vscode-extension-${{ matrix.arch }}-${{ github.run_id }}
+          path: "vscode/*.vsix"
```

Comment:
I like that too. I will update the workflow. https://github.com/marvinpinto/action-automatic-releases has been archived. I will look into alternatives.

Reply:
I tried two other alternatives but it is not updating the artifacts/commit. The alternative will be to create our scripts to remove the artifacts and then upload (marvinpinto's script: https://github.com/marvinpinto/actions/blob/master/scripts/mirror-actions.sh) 
@djzager which path do you suggest I pursue? use the archived action or write our own. I can write our own since I am not a fan of using unmaintained code.

Reply:
We use https://github.com/ncipollo/release-action for the Konveyor release workflows...still looks like a legit process. It seems that this is valid and should work to maintain a single/rolling `latest` release (yes it is AI generated) :-D

```
      - name: Create or update a GitHub release
        uses: ncipollo/release-action@v1
        with:
          tag: latest
          name: Latest Release
          draft: false
          prerelease: false
          body: "Rolling release of the latest build from the main branch."
          files: |
            path-to-your-artifact-1
            path-to-your-artifact-2
          token: ${{ secrets.GITHUB_TOKEN }}
```

Reply:
I tried this one. It only updates the .vsix file and not the source code zip files. I also tried another one `softprops/action-gh-release@v1` which did the same thing.

Reply:
I will try again probably tomorrow. I might have missed something for sure.

Reply:
Did you make use of `allowUpdates`?
> An optional flag which indicates if we should update a release if it already exists. Defaults to false.

I notice also:

removeArtifacts | Indicates if existing release artifacts should be removed. | false | false
-- | -- | -- | --
replacesArtifacts | Indicates if existing release artifacts should be replaced. | false | true

You may well have tried these already but we can talk about it more tomorrow. We may have to delete the release and recreate it if it is going to be difficult.

Reply:
I used both `allowUpdates` and `replacesArtifacts`. It only replaces the `.vsix` package and not the source code

Reply:
@djzager I have updated the code to cut a prerelease for every merge using a name that is `dev-<commit-id_short>`. We can add a cleanup script or if this doesnt workout then I can work on the other approach of cleaning up latest before release. Let me know what you think

Reply:
eg: https://github.com/savitharaghunathan/editor-extensions/releases/tag/dev-4c084ab5